File: codility/Peaks.js
function solution(A) {
    const len = A.length;
    const pArr = [];
    for (let i = 1; i < len - 1; i++) {
        if (A[i - 1] < A[i] && A[i] > A[i + 1]) {
            pArr.push(i);
        }
    }
    let n = pArr.length;
    if (n === 0) {
        return 0;
    }
    while (!calc() && n > 1) {
        n--;
    }
    function calc() {
        if (len % n !== 0) {
            return false;
        }
        const set = new Set();
        const groupNum = len / n;
        for (let i = 0; i < pArr.length; i++) {
            const index = pArr[i];
            set.add(Math.floor(index / groupNum));
        }
        return set.size === n;
    }
    return n;
}

File: codility/python/Peaks.py
import math

def solution(A):
    # Implement your solution here
    peaks = []
    N = len(A)
    for i in range(1, N-1):
        if A[i] > max(A[i-1], A[i+1]):
            peaks.append(i)
    n = len(peaks)
    if n == 0:
        return 0
    for i in range(n, 0, -1):
        if N % i != 0:
            continue
        groups = int(N / i)
        my_set = set()
        for j in range(n):
            a = math.floor(peaks[j] / groups)
            my_set.add(a) 
        if len(my_set) == i:
            return i

    return 1